courses/signals.py

```diff
@@ -1,15 +1,40 @@
 from django.dispatch import receiver
 
+from student.signals import ENROLL_STATUS_CHANGE
+from student.models import EnrollStatusChange
 from xmodule.modulestore.django import SignalHandler
 
 
 @receiver(SignalHandler.course_published, dispatch_uid='fun.courses.signals.update_courses')
 def update_course_meta_data_on_studio_publish(sender, course_key, **kwargs):
+    """Trigger hook when publishing a change to a course"""
     from django.conf import settings
     if getattr(settings, "COURSE_SIGNALS_DISABLED", False):
         return 'FUN courses meta data update has been skipped.'
 
     from .tasks import update_courses_meta_data
-    # course_key is a CourseKey object and course_id its sting representation
+    # course_key is a CourseKey object and course_id its string representation
     update_courses_meta_data.delay(course_id=unicode(course_key))
-    return 'FUN courses meta data update has been triggered.'
+    return 'FUN courses meta data update has been triggered from course publish.'
+
+
+@receiver(
+    ENROLL_STATUS_CHANGE,
+    dispatch_uid='fun.courses.signals.change_enrollment_status')
+def sync_openedx_to_richie_after_enrollment_status_change(
+        sender, event=None, course_id=None, **kwargs
+    ):
+    """
+    Trigger hook when changing a course enrollment and the change if of the type "enroll
+    """
+
+    if event != EnrollStatusChange.enroll:
```

Comment:
WDYT to trigger it also on unenroll to keep stats fresh as possible?
```suggestion
    if event not in [EnrollStatusChange.enroll, EnrollStatusChange.unenroll]:
```

Reply:
I thought of it but we don't count Enrollment objects of the enroll type, we also count unenrolled as "already enrolled"... which makes sense because we are boasting how many people enrolled to our course not how many are still enrolled at the time we speak?

Reply:
Okay ! 
It is not a big point, in fact information is just delayed.